diff --git a/packages/autoloader/src/lazyLoad.ts b/packages/autoloader/src/lazyLoad.ts
index 2e96f1a..d3bfe2a 100644
--- a/packages/autoloader/src/lazyLoad.ts
+++ b/packages/autoloader/src/lazyLoad.ts
@@ -1,7 +1,7 @@
 import { resolveLoader } from "./resoleveLoader";
 import { failedTags, loadingTags } from "./tags";
 import { IConfig, ILoader, INameSpaceInfo, IPrefixMap, ITagInfo } from "./types";
@@
-const observedCustomElements: WeakSet<Element> = new Set();
+const observedCustomElements: WeakSet<Element> = new WeakSet<Element>();
