diff --git a/packages/router/src/components/Layout.ts b/packages/router/src/components/Layout.ts
index 3f2f7c0..60f3d3d 100644
--- a/packages/router/src/components/Layout.ts
+++ b/packages/router/src/components/Layout.ts
@@ -7,7 +7,6 @@ const cache = new Map<string, string>();
 
 export class Layout extends HTMLElement implements ILayout {
   private _uuid: string = getUUID();
-  private _name: string = '';
   private _initialized: boolean = false;
   constructor() {
     super();
   }
@@
   get name(): string {
-    return this._name;
+    // Layout 要素が DOM に挿入されないケース（parseで置換）でも name を取れるようにする
+    return this.getAttribute('name') || '';
   }
 
   private _initialize() {
-    this._name = this.getAttribute('name') || '';
     this._initialized = true;
   }
